src/hooks/responsive.ts

```diff
@@ -11,3 +13,35 @@ export const useIsSmDown = () => useBreakpointDown('sm');
 export const useIsMdDown = () => useBreakpointDown('md');
 export const useIsLgDown = () => useBreakpointDown('lg');
 export const useIsXlDown = () => useBreakpointDown('xl');
+
+export const useBreakpointHideDown = (breakpoint: Breakpoint) => {
+  const theme = useTheme();
+
+  return css`
+    ${theme.breakpoints.down(breakpoint)} {
+      display: none;
+    }
+  `;
+};
+
+export const useHideXsDownCss = () => useBreakpointHideDown('xs');
```

Comment:
I don't see where these are used. I also don't think we want to prefer using  hooks for responsiveness

Reply:
For now only a few are used, but they are util hooks that can be used in any component.

They declare styles the way we currently do (using hooks and Emotions' `css` function), so as a result only CSS is output. The reason I created them is so the DOM can be a bit more explicit when it comes to defining what elements show on what screen dimensions and to avoid having to duplicate CSS. If you look in the codebase, there's currently quite a few places declaring the same CSS, something like:
```
css`
  display: block;
 
  ${theme.breakpoints.down('md')} {
    display: none;
  }
`
```
Using these hooks, we don't have to write down this logic for each component and the DOM becomes easier to read in my opinion, since you'll have something like this:

```jsx
<ElementAlwaysVisible />

<ElementVisibleOnDesktopOnly css={[styles.customStyling, hideXlDownCss]} />

<ElementVisibleOnMobileOnly css={[styles.customStyling, showXsDownCss]} />
```

Reply:
Yeah, this is a downside of css in js. It isn't as easy to compose classes. Is this the best way to reduce the duplicated code though, because the tradeoff is adding event listeners to the dom. 

One alternative would be to assign the `css` block in your example to a variable and importing and interpolating that variable.

Reply:
I conflated these hooks with the ones that do use listeners. Looks like these don't. I like this.